luigi/contrib/sqla.py

```diff
@@ -152,8 +150,21 @@ def __init__(self, connection_string, target_table, update_id, echo=False):
         """
         self.target_table = target_table
         self.update_id = update_id
-        self.engine = sqlalchemy.create_engine(connection_string, echo=echo)
+        self.connection_string = connection_string
+        self.echo = echo
         self.marker_table_bound = None
+        self._engine = None
+        self._mutex = Lock()

```

Comment:
Also, now that I think about it: Luigi doesn't use threads, so I'm not sure if you really need a Mutex.